src/main/java/zyenyo/Database.java

```diff
@@ -80,9 +80,16 @@ public static double addTest(long discordId, double wpm, double accuracy, double
 				.append("tp", tp)
 				.append("date", LocalDateTime.now())
 				);
-
-		return getWeightedTp(discordId) - initialWeightedTp;
-
+		
+		double newWeightedTp = getWeightedTp(discordId);
+		
+		users.updateMany(
```

Comment:
you can use `updateOne` here instead of updateMany. there likely isn't gonna be another user with the same id so this might save some performance